### DISCUSSION TITLE
Removal of some deprecated code in 5.0

### DISCUSSION THREAD

**Discussion by HLeithner** [joomla/joomla-cms#40376] [5. Implemented Features]
As per Production Department motion we try hard to make Joomla 5.0 compatible with 4.4. Never the less we have code which we could and should remove without a big impact on the Joomla eco system. Things like left over files which aren't used since ages or functions in components which are not part of our b/c promise but declared to be keep till 6.0 and not used anymore.

Here I will maintain a list of code that we would like to remove in 5.0 and voted in production as exception to the b/c motion.

If you want to discuss parts of the deprecation please make or continue only one thread per item.

### Remove library/src/Schema/ChangeItem/SqlsrvChangeItem.php 
The MS SQL is not supported since 4.0 and this file is a left over without any use. (already removed in 5.0 for now, can be reverted if there is a good reason)

### Remove of administrator/compoonents/com_menus/src/Controller/MenusController.php method resync 
This function was only used in the Joomla 1.6 upgrade and is a left over. keeping it makes no sense because not used. Additionally it isn't part of the b/c policy.

### Upgrade Webauthnlib to Version 3 
This has already been done in Joomla 5.0 because we need to have an uptodate version of the upstream webauthn library for security reasons. Maintaining our own version doesn't make sense here especially by the expected impact this update has for 3rd party developer. It is only used by our own webauthn plugin, extensions using this may have to be updated.

As part of the Webauthn library upgrade the following files will be removed because they are no longer needed
* plugins/multifactorauth/webauthn/src/Hotfix
* plugins/system/webauthn/src/Hotfix

As part of the Webauthn library ugprade we need to upgrade our PSR Logger dependency which effects extensions which extends our Delegating PSR Logger. It was never meant to be extended and has been changed to final. Beside that the function signature has been more strict now to be compatible with upstream (iirc the parameter definition was already in the docblock).

# Approval Information
Approved: 25.02.2026
Minimum Target Version: 7.0

**Comment by richard67** [comment]
> Here I will maintain a list of code that we would like to remove in 5.0 and voted in production as exception to the b/c motion.
>
> If you want to discuss parts of the deprecation please make or continue only one thread per item.
>
> Remove library/src/Schema/ChangeItem/SqlsrvChangeItem.php

The "library/src/Schema/ChangeItem/SqlsrvChangeItem.php" file has already been removed in the 5.0-dev branch with my PR #38405 .

So if during the discussion it turns out that it should be restored, it will need to revert that PR.

But you should know that this library was only used by the core to check the database structure of the core with the database checker/fixer, and the core can't be installed on a MS SQL Server or Azure database since 4.0.0. The support to connect to such databases with our database package of the framework is not concerned by the removal of that file.

**Reply by HLeithner** [reply]
I added that information to the message above

**Comment by rdeutz** [comment]
The Joomla Maintainer Teams decided on the 25.02.2025 meeting to accept the feature.

**Comment by HLeithner** [comment]
Already done in earlier version, thanks for the heads up @laoneo
